diff -up kate-20.03.90/addons/gdbplugin/configview.cpp.omv~ kate-20.03.90/addons/gdbplugin/configview.cpp
--- kate-20.03.90/addons/gdbplugin/configview.cpp.omv~	2020-04-11 23:55:09.009641071 +0200
+++ kate-20.03.90/addons/gdbplugin/configview.cpp	2020-04-11 23:55:13.072689278 +0200
@@ -118,11 +118,7 @@ ConfigView::ConfigView(QWidget *parent,
     m_advanced->hide();
 
     connect(m_targetCombo, &QComboBox::editTextChanged, this, &ConfigView::slotTargetEdited);
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_targetCombo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &ConfigView::slotTargetSelected);
-#else
-    connect(m_targetCombo, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, &ConfigView::slotTargetSelected);
-#endif
     connect(m_addTarget, &QToolButton::clicked, this, &ConfigView::slotAddTarget);
     connect(m_copyTarget, &QToolButton::clicked, this, &ConfigView::slotCopyTarget);
     connect(m_deleteTarget, &QToolButton::clicked, this, &ConfigView::slotDeleteTarget);
diff -up kate-20.03.90/addons/gdbplugin/plugin_kategdb.cpp.omv~ kate-20.03.90/addons/gdbplugin/plugin_kategdb.cpp
--- kate-20.03.90/addons/gdbplugin/plugin_kategdb.cpp.omv~	2020-04-11 23:54:10.873954669 +0200
+++ kate-20.03.90/addons/gdbplugin/plugin_kategdb.cpp	2020-04-11 23:54:14.853001444 +0200
@@ -138,11 +138,7 @@ KatePluginGDBView::KatePluginGDBView(KTe
     m_stackTree->setAutoScroll(false);
     connect(m_stackTree, &QTreeWidget::itemActivated, this, &KatePluginGDBView::stackFrameSelected);
 
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_threadCombo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &KatePluginGDBView::threadSelected);
-#else
-    connect(m_threadCombo, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, &KatePluginGDBView::threadSelected);
-#endif
 
     m_localsView = new LocalsView();
 
diff -up kate-20.03.90/addons/katesql/katesqlview.cpp.omv~ kate-20.03.90/addons/katesql/katesqlview.cpp
--- kate-20.03.90/addons/katesql/katesqlview.cpp.omv~	2020-04-11 23:55:22.293798796 +0200
+++ kate-20.03.90/addons/katesql/katesqlview.cpp	2020-04-11 23:55:33.796935624 +0200
@@ -90,14 +90,7 @@ KateSQLView::KateSQLView(KTextEditor::Pl
     connect(m_manager, &SQLManager::queryActivated, this, &KateSQLView::slotQueryActivated);
     connect(m_manager, &SQLManager::connectionCreated, this, &KateSQLView::slotConnectionCreated);
     connect(m_manager, &SQLManager::connectionAboutToBeClosed, this, &KateSQLView::slotConnectionAboutToBeClosed);
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_connectionsComboBox, QOverload<const QString &>::of(&QComboBox::currentIndexChanged), this, &KateSQLView::slotConnectionChanged);
-#else
-    connect(m_connectionsComboBox, QOverload<int, const QString &>::of(&QComboBox::currentIndexChanged), this, [this](int index, const QString &connection) {
-        Q_UNUSED(index);
-        slotConnectionChanged(connection);
-    });
-#endif
     stateChanged(QStringLiteral("has_connection_selected"), KXMLGUIClient::StateReverse);
 }
 
diff -up kate-20.03.90/addons/project/kateprojectinfoviewcodeanalysis.cpp.omv~ kate-20.03.90/addons/project/kateprojectinfoviewcodeanalysis.cpp
--- kate-20.03.90/addons/project/kateprojectinfoviewcodeanalysis.cpp.omv~	2020-04-11 23:53:56.826789802 +0200
+++ kate-20.03.90/addons/project/kateprojectinfoviewcodeanalysis.cpp	2020-04-11 23:54:01.888849168 +0200
@@ -67,11 +67,7 @@ KateProjectInfoViewCodeAnalysis::KatePro
      * Connect selection change callback
      * and attach model to code analysis selector
      */
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_toolSelector, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &KateProjectInfoViewCodeAnalysis::slotToolSelectionChanged);
-#else
-    connect(m_toolSelector, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, &KateProjectInfoViewCodeAnalysis::slotToolSelectionChanged);
-#endif
     m_toolSelector->setModel(KateProjectCodeAnalysisSelector::model(this));
 
     /**
diff -up kate-20.03.90/addons/project/kateprojectpluginview.cpp.omv~ kate-20.03.90/addons/project/kateprojectpluginview.cpp
--- kate-20.03.90/addons/project/kateprojectpluginview.cpp.omv~	2020-04-11 23:53:42.237619011 +0200
+++ kate-20.03.90/addons/project/kateprojectpluginview.cpp	2020-04-11 23:53:46.503668906 +0200
@@ -83,11 +83,7 @@ KateProjectPluginView::KateProjectPlugin
     m_stackedProjectViews = new QStackedWidget(m_toolView);
     m_stackedProjectInfoViews = new QStackedWidget(m_toolInfoView);
 
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_projectsCombo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &KateProjectPluginView::slotCurrentChanged);
-#else
-    connect(m_projectsCombo, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, &KateProjectPluginView::slotCurrentChanged);
-#endif
     connect(m_reloadButton, &QToolButton::clicked, this, &KateProjectPluginView::slotProjectReload);
 
     /**
diff -up kate-20.03.90/addons/search/plugin_search.cpp.omv~ kate-20.03.90/addons/search/plugin_search.cpp
--- kate-20.03.90/addons/search/plugin_search.cpp.omv~	2020-04-11 23:54:23.618104587 +0200
+++ kate-20.03.90/addons/search/plugin_search.cpp	2020-04-11 23:54:35.790248066 +0200
@@ -419,16 +419,8 @@ KatePluginSearchView::KatePluginSearchVi
     connect(m_ui.searchButton, &QPushButton::clicked, this, &KatePluginSearchView::startSearch);
 
     connect(m_ui.displayOptions, &QToolButton::toggled, this, &KatePluginSearchView::toggleOptions);
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_ui.searchPlaceCombo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &KatePluginSearchView::searchPlaceChanged);
-#else
-    connect(m_ui.searchPlaceCombo, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, &KatePluginSearchView::searchPlaceChanged);
-#endif
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_ui.searchPlaceCombo, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, [this](int) {
-#else
-    connect(m_ui.searchPlaceCombo, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, [this](int) {
-#endif
         if (m_ui.searchPlaceCombo->currentIndex() == Folder) {
             m_ui.displayOptions->setChecked(true);
         }
diff -up kate-20.03.90/kate/kateconfigdialog.cpp.omv~ kate-20.03.90/kate/kateconfigdialog.cpp
--- kate-20.03.90/kate/kateconfigdialog.cpp.omv~	2020-04-11 23:54:42.538327728 +0200
+++ kate-20.03.90/kate/kateconfigdialog.cpp	2020-04-11 23:55:00.891544837 +0200
@@ -162,11 +162,7 @@ KateConfigDialog::KateConfigDialog(KateM
     m_cmbQuickOpenMatchMode->addItem(i18n("Filepath"), QVariant(KateQuickOpenModel::Columns::FilePath));
     m_cmbQuickOpenMatchMode->setCurrentIndex(m_cmbQuickOpenMatchMode->findData(m_mainWindow->quickOpenMatchMode()));
     m_mainWindow->setQuickOpenMatchMode(m_cmbQuickOpenMatchMode->currentData().toInt());
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_cmbQuickOpenMatchMode, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &KateConfigDialog::slotChanged);
-#else
-    connect(m_cmbQuickOpenMatchMode, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, &KateConfigDialog::slotChanged);
-#endif
     vbox->addLayout(hlayout);
     // quick open list mode
     hlayout = new QHBoxLayout;
@@ -179,11 +175,7 @@ KateConfigDialog::KateConfigDialog(KateM
     m_cmbQuickOpenListMode->addItem(i18n("All Projects Files"), QVariant(KateQuickOpenModel::List::AllProjects));
     m_cmbQuickOpenListMode->setCurrentIndex(m_cmbQuickOpenListMode->findData(m_mainWindow->quickOpenListMode()));
     m_mainWindow->setQuickOpenListMode(static_cast<KateQuickOpenModel::List>(m_cmbQuickOpenListMode->currentData().toInt()));
-#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
     connect(m_cmbQuickOpenListMode, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this, &KateConfigDialog::slotChanged);
-#else
-    connect(m_cmbQuickOpenListMode, static_cast<void (QComboBox::*)(int, const QString &)>(&QComboBox::currentIndexChanged) , this, &KateConfigDialog::slotChanged);
-#endif
     vbox->addLayout(hlayout);
     layout->addWidget(buttonGroup);
 
